query-service-impl/src/main/java/org/hypertrace/core/query/service/AdditionalHandlerFiltersConfig.java

```diff
@@ -0,0 +1,128 @@
+package org.hypertrace.core.query.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.util.JsonFormat;
+import com.typesafe.config.Config;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.experimental.NonFinal;
+import lombok.extern.slf4j.Slf4j;
+import org.hypertrace.core.query.service.api.Filter;
+import org.hypertrace.core.query.service.api.Operator;
+
+@Slf4j
+public class AdditionalHandlerFiltersConfig {
+  private static final String ADDITIONAL_TENANT_FILTERS_CONFIG_KEY = "additionalTenantFilters";
+  private final Map<String, List<Filter>> tenantToAdditionalFiltersMap;
+
+  public AdditionalHandlerFiltersConfig(Config config, Optional<String> timeFilterColumnName) {
+    if (config.hasPath(ADDITIONAL_TENANT_FILTERS_CONFIG_KEY)) {
+      this.tenantToAdditionalFiltersMap =
+          config.getConfigList(ADDITIONAL_TENANT_FILTERS_CONFIG_KEY).stream()
+              .map(filterConfig -> new TenantFilters(filterConfig, timeFilterColumnName))
+              .collect(
+                  Collectors.toMap(
+                      tenantFilters -> tenantFilters.tenantId,
+                      tenantFilters -> tenantFilters.filters));
+    } else {
+      this.tenantToAdditionalFiltersMap = Collections.emptyMap();
+    }
+  }
+
+  public List<Filter> getAdditionalFiltersForTenant(String tenantId) {
```

Comment:
We should definitely add test cases for this new class `AdditionalHandlerFiltersConfig`.

Reply:
All scenarios for this config is already covered in PinotHanderTest since we're taking input from application.conf there.

Reply:
Added config tests too